python/ray/data/_internal/numpy_support.py

```diff
@@ -72,16 +72,25 @@ def convert_udf_returns_to_numpy(udf_return_col: Any) -> Any:
                 udf_return_col = [np.array(e) for e in udf_return_col]
             shapes = set()
             has_object = False
+            dtype = None
             for e in udf_return_col:
                 if isinstance(e, np.ndarray):
                     shapes.add((e.dtype, e.shape))
+                elif isinstance(e, bytes):
+                    # Don't convert variable length binary data to Numpy arrays as it
+                    # treats zero encoding as termination by default.
+                    # Per recommendation from
+                    # https://github.com/apache/arrow/issues/26470,
+                    # we use object dtype.
+                    # https://github.com/ray-project/ray/issues/35586#issuecomment-1558148261
+                    dtype = object
```

Comment:
Would be simpler to set has_object=True. All create ragged ndarray does is create an object array under the hood.